### PR TITLE
xDS plugin is going to use LRS stream to report load to balancer. Remove

### PR DESCRIPTION
**Pull request by vishalpowar**
the current grpclb specific load reporting from the implementation.

The changes here also mean that the plugin will just get the initial
response and no subsequent response from the balancer.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,969      Total (<)      2,020,494

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,175,347      Total (<)     11,175,623

 No significant differences in binary sizes


```

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 1]
```diff
@@ -668,42 +668,26 @@ bool XdsLb::BalancerCallState::LoadReportCountersAreZero(
...
   // Construct message payload.
   GPR_ASSERT(send_message_payload_ == nullptr);
   xds_grpclb_request* request =
       xds_grpclb_load_report_request_create_locked(client_stats_.get());
+
   // Skip client load report if the counters were all zero in the last
   // report and they are still zero in this one.
   if (LoadReportCountersAreZero(request)) {
     if (last_client_load_report_counters_were_zero_) {
-      xds_grpclb_request_destroy(request);
```
Comment (AspirinSJL) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Why should we remove this line?

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 2]
```diff
@@ -668,42 +668,26 @@ bool XdsLb::BalancerCallState::LoadReportCountersAreZero(
          (drop_entries == nullptr || drop_entries->empty());
 }
 
+// TODO(vpowar): Use LRS to send the client Load Report.
 void XdsLb::BalancerCallState::SendClientLoadReportLocked() {
   // Construct message payload.
   GPR_ASSERT(send_message_payload_ == nullptr);
   xds_grpclb_request* request =
       xds_grpclb_load_report_request_create_locked(client_stats_.get());
+
```
Comment (AspirinSJL) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Please remove blank lines in functions. We prefer comments than blank lines.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc [hunk 3]
```diff
@@ -668,42 +668,26 @@ bool XdsLb::BalancerCallState::LoadReportCountersAreZero(
...
-      grpc_raw_byte_buffer_create(&request_payload_slice, 1);
-  grpc_slice_unref_internal(request_payload_slice);
+  // TODO(vpowar): Send the report on LRS stream.
   xds_grpclb_request_destroy(request);
-  // Send the report.
-  grpc_op op;
-  memset(&op, 0, sizeof(op));
-  op.op = GRPC_OP_SEND_MESSAGE;
-  op.data.send_message.send_message = send_message_payload_;
-  GRPC_CLOSURE_INIT(&client_load_report_closure_, ClientLoadReportDoneLocked,
```
Comment (AspirinSJL) [inline src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Please remove the `ClientLoadReportDoneLocked` definition to fix the test failure.

File: src/core/ext/filters/client_channel/lb_policy/xds/xds.cc (hunk 1 above)
Comment (vishalpowar) [reply src/core/ext/filters/client_channel/lb_policy/xds/xds.cc]: Ahh, missed it. Removed

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++

 -------------- SHRINKING                                                                            --------------
  -0.0%    -152 [None]                                                                               -2.38Ki  -0.0%
  -3.0%    -480 src/core/ext/filters/client_channel/lb_policy/xds/xds.cc                                -480  -3.0%
     -65.7%    -369 grpc_core::(anonymous namespace)::XdsLb::BalancerCallState::SendClientLoadReportLock    -369 -65.7%
      [DEL]    -100 grpc_core::(anonymous namespace)::XdsLb::BalancerCallState::ClientLoadReportDoneLock    -100  [DEL]
      -2.8%     -11 [Unmapped]                                                                               -11  -2.8%

  -0.0%    -632 TOTAL                                                                                -2.84Ki  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,019,973      Total (<)      2,020,494

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,175,342      Total (<)     11,175,632

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by AspirinSJL** [comment]
Please squash the commits. I will then approve the PR. Thanks!

**Pull request merged**